src/ConstructiveSolidGeometry/VolumePrimitives/Ellipsoid.jl

```diff
@@ -47,19 +47,32 @@ This is a sphere with inner radius 1 and outer radius 2.
 
 See also [Constructive Solid Geometry (CSG)](@ref).
 """
-@with_kw struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
-    r::TR = 1
-    φ::TP = nothing
-    θ::TT = nothing
+struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
+    r::TR 
+    φ::TP 
+    θ::TT 
+
+    origin::CartesianPoint{T}
+    rotation::SMatrix{3,3,T,9}
+        
+end
+
+function Ellipsoid{T,CO}(r::TR,φ::TP,θ::TT,origin::CartesianPoint{T}=zero(CartesianPoint{T}),
+    rotation::SMatrix{3,3,T,9}=one(SMatrix{3, 3, T, 9})) where {T,CO,TR<:T,TP<:Nothing,TT<:Nothing} 
+    Ellipsoid{T,CO,TR,TP,TT}(r,φ,θ,origin,rotation)
+end
```

Comment:
I think keywords would be better here:

```julia
function Ellipsoid{T,CO}(;
    r::TR = 1,
    origin = zero(CartesianPoint{T}),
    rotation = one(SMatrix{3, 3, T, 9})
) where {T,CO,TR} 
    Ellipsoid{T,CO,TR,Nothing,Nothing}(r,nothing,nothing,origin,rotation)
end
```

I think we can drop the type annotations for `origin` and `rotation` here. 
We should add `convert` methods for e.g. `CartesianPoint` if we don't already have them
so Julia will automatically convert them to the correct type with elements of type `T` in the default constructor.